x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/last_value.tsx

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiFormRow, EuiSelect, EuiButtonGroup } from '@elastic/eui';
+import { OperationDefinition } from './index';
+import { FieldBasedIndexPatternColumn } from './column_types';
+import { IndexPatternField, IndexPattern } from '../../types';
+import { updateColumnParam } from '../layer_helpers';
+import { DataType } from '../../../types';
+
+function ofName(name: string) {
+  return i18n.translate('xpack.lens.indexPattern.lastValueOf', {
+    defaultMessage: 'Last value of {name}',
+    values: { name },
+  });
+}
+
+const supportedTypes = new Set(['string', 'boolean', 'number', 'ip']);
+
+function getDateFields(indexPattern: IndexPattern): IndexPatternField[] {
+  if (indexPattern.timeFieldName) {
+    return [
+      ...indexPattern.fields.filter((field) => field.name === indexPattern.timeFieldName),
+      ...indexPattern.fields.filter(
+        (field) => field.type === 'date' && indexPattern.timeFieldName !== field.name
+      ),
+    ];
+  } else {
+    return indexPattern.fields.filter((field) => field.type === 'date');
+  }
```

Comment:
nit: rather than filter multiple times, it can be filtered once and then sorted:

```js
function getDateFields(indexPattern: IndexPattern): IndexPatternField[] {
    const dateFields = indexPattern.fields.filter((field) => field.type === 'date');
    if(indexPattern.timeFieldName){
        dateFields.sort( ({nameA, nameB}) => {
            if(nameA === indexPattern.timeFieldName){
                return -1;
            }
            if(nameB === indexPattern.timeFieldName){
                return 1;
            }
            return 0;
        });
    }
    return dateFields.
}
```

This should also streamline the flow